modules/mobile/app/controllers/mobile/v0/translations_controller.rb

```diff
@@ -25,27 +25,12 @@ def file
         Rails.root.join('modules', 'mobile', 'app', 'assets', 'translations', 'en', 'common.json')
       end
 
-      def file_last_changed
-        @file_last_changed ||= begin
-          # can't interpolate file path into system call because it violates a security rule
-          timestamp = `git log -1 --format='%ct' -- modules/mobile/app/assets/translations/en/common.json`
-          Integer(timestamp)
-        end
+      def file_md5
+        @file_md5 ||= Digest::MD5.file(file).hexdigest
       end
 
       def needs_translations?
-        params[:current_version].nil? || (current_version < file_last_changed)
-      end
-
-      def current_version
-        return nil if params[:current_version].nil?
-
-        Integer(params[:current_version])
-      rescue ArgumentError
-        raise Common::Exceptions::UnprocessableEntity.new(
-          detail: "#{params[:current_version]} is not an integer",
-          source: self.class.to_s
-        )
+        params[:current_version] != file_md5
```

Comment:
The idea is that the app will only get the translation file from this endpoint, it won't start with a hardcoded one on the app itself, right? I'm trying to think of a scenario where the app has a newer version of the translation file then vets-api.

Reply:
Also, the prepackaged version the app comes with will also come from this endpoint. The FE build process will ping this endpoint without a query param in order to get the translations file.